web/src/navigator.rs

```diff
@@ -60,39 +44,30 @@ impl NavigatorBackend for WebNavigatorBackend {
         window_spec: Option<String>,
         vars_method: Option<(NavigationMethod, IndexMap<String, String>)>,
     ) {
-        const JAVASCRIPT_PREFIX: &str = "javascript:";
+        const JAVASCRIPT_PREFIX: &str = "javascript";
 
         if let Some(window) = window() {
-            let url_trimmed = url.trim();
-            if url_trimmed
-                .get(..JAVASCRIPT_PREFIX.len())
-                .unwrap_or_default()
-                .eq_ignore_ascii_case(JAVASCRIPT_PREFIX)
-            {
-                let target = window_spec.unwrap_or_else(|| "".to_string());
-                if target.is_empty() || target == "_self" || target == "undefined" {
-                    self.run_script(&url);
+            let url = match Url::parse(&url) {
+                Ok(parsed) => parsed,
+                Err(ParseError::RelativeUrlWithoutBase) => {
+                    // If the url has no base, create a base for the url
+                    let base = Url::parse(&window.location().href().unwrap()).unwrap();
+                    base.join(&url).unwrap()
                 }
-                return;
-            }
-
-            let window_url = if url.is_empty() {
-                "".to_string()
-            } else if url_trimmed.is_empty() {
-                "./".to_string()
-            } else {
-                url.to_string()
+                _ => return,
             };
 
-            let form_url = if let Ok(parsed_url) = Url::parse(&url) {
-                self.pre_process_url(parsed_url).to_string()
-            } else {
-                url.to_string()
-            };
+            // If allowScriptAccess is disabled, we should reject the javascript scheme
+            if !self.allow_script_access && url.scheme() == JAVASCRIPT_PREFIX {
+                log::error!("The javascript scheme is not supported");
```

Comment:
Why logging an error in this case? Reaching here is not a bad thing.

Reply:
Maybe a warning would be better? Flash doesn't log anything, but I think it might be helpful for debugging to log something